packages/hardhat-plugin/src/type-extensions.ts

```diff
@@ -2,28 +2,25 @@
 import "hardhat/types/config";
 import "hardhat/types/runtime";
 
-import type { IgnitionConfig } from "./index";
+import { DeployConfig } from "@ignored/ignition-core";
 
 import { IgnitionHelper } from "./ignition-helper";
 
 declare module "hardhat/types/config" {
-  // TODO: double check each of these makes sense
   export interface ProjectPathsUserConfig {
     ignition?: string;
-    deployments?: string;
   }
 
   export interface ProjectPathsConfig {
     ignition: string;
-    deployments: string;
   }
 
   export interface HardhatUserConfig {
-    ignition?: Partial<IgnitionConfig>;
+    ignition?: Partial<DeployConfig>;
   }
 
   export interface HardhatConfig {
-    ignition: IgnitionConfig;
+    ignition: Partial<DeployConfig>;
```

Comment:
Note that https://github.com/NomicFoundation/ignition/pull/353/files#r1279757963 may give the impression that we should resolve things here. It's not mandatory. This looks good.